chart/templates/hooks.yaml

```diff
@@ -0,0 +1,76 @@
+# generate helm hooks from contents in the `hooks` folder
+{{ range $_, $hook := tuple "pre-delete" "pre-upgrade" }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{$hook}}-helm-hook
+  namespace: {{ $.Release.Namespace | quote }}
+  labels:
+    app.kubernetes.io/instance: {{ $.Release.Name | quote }}
+    app.kubernetes.io/managed-by: {{ $.Release.Service | quote }}
+  annotations:
+    "helm.sh/hook": {{$hook}}
+    "helm.sh/hook-weight": "-2"
+    "helm.sh/hook-delete-policy": before-hook-creation
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ $.Release.Name }}-{{$hook}}-hook"
+  labels:
+    app.kubernetes.io/managed-by: {{ $.Release.Service | quote }}
+    app.kubernetes.io/instance: {{ $.Release.Name | quote }}
+  annotations:
+    "helm.sh/hook": {{$hook}}
+    "helm.sh/hook-weight": "-1"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{ $.Release.Name }}"
+      labels:
+        app.kubernetes.io/managed-by: {{ $.Release.Service | quote }}
+        app.kubernetes.io/instance: {{ $.Release.Name | quote }}
+        helm.sh/chart: {{ include "kubecf.chart" $ }}
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{$hook}}-helm-hook
+      containers:
+      - name: {{$hook}}-job
+        image: {{ $.Values.hooks.image | quote }}
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        command: ["/bin/bash","-cex"]
+        args: 
+        - |
+           for f in /hooks/*.sh; do 
+            bash "$f" -H || break
```

Comment:
As far as I can tell, the `-H` is being passed to the script, which promptly ignores it? If it was meant for `bash`, it would be before the file name, and be the same as `set -o histexpand`, which… makes no sense?

I feel like I'm just dumb today…